### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by misha1sh**
Setup instructions extended

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
 
 Setup
 -----
+shaderc-rs needs [shaderc library](https://github.com/google/shaderc) be installed
```
Comment (antiagainst) [inline README.md]: Could you end the sentence with a period?

File: README.md [hunk 2]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+
+shaderc library can be obtained from [here](https://github.com/google/shaderc#downloads)
+and linked, or can be [built from source](#building-from-source):
+
+1. Setup, using static linking
```
Comment (antiagainst) [inline README.md]: s/.../Linking with pre-built static library/

File: README.md [hunk 3]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+    - On Linux, If the `SHADERC_LIB_DIR` environment variable is set to
+      `/path/to/shaderc/libs/`, it will take precedence and `libshaderc_combined.a`
+      (and the glsang and SPIRV libraries) will be searched in the
+      `/path/to/shaderc/libs/` directory. 
+    - On Linux, `/usr/lib/` will be automatically searched for `libshaderc_combined.a`
+      if `SHADERC_LIB_DIR` is not setted.
+    - On Windows MSVC, If the `SHADERC_LIB_DIR` environment variable is set to
+      `/path/to/shaderc/libs/`, `shaderc_combined.lib` will be searched in the
+      `/path/to/shaderc/libs/` directory
+2. Setup, using dynamic linking
```
Comment (antiagainst) [inline README.md]: Linking with pre-built dynamic library

File: README.md [hunk 4]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
-2. If the `SHADERC_LIB_DIR` environment variable is set to
-   `/path/to/shaderc/libs/`, it will take precedence and `libshaderc_combined.a`
-   (and the glsang and SPIRV libraries on Linux) will be searched in the
-   `/path/to/shaderc/libs/` directory.
-3. On Linux, `/usr/lib/` will be automatically searched for system libraries
-   if none of the above were given.
-4. If no other option was set or succeeded, shaderc-sys will fall back to
-   checking out and compiling a copy of shaderc.  This procedure is quite slow.
+1. Building from source, if option `--features build-from-source` is used.
+2. If the SHADERC_LIB_DIR environment variable is set to `/path/to/shaderc/libs/`,
```
Comment (antiagainst) [inline README.md:105]: Quote SHADERC_LIB_DIR with bckticks.

File: README.md [hunk 5]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+   `/path/to/shaderc/libs/` will be searched for static and dynamic libs.
```
Comment (antiagainst) [inline README.md]: static or dynamic

File: README.md [hunk 6]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+3. On Linux, `/usr/lib/` will be searched for static and dynamic libs
+   if the SHADERC_LIB_DIR is not setted
```
Comment (antiagainst) [inline README.md]: Quote SHADERC_LIB_DIR with bckticks. 

"... not set"

And end with a period.

File: README.md [hunk 7]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+      `/path/to/shaderc/libs/` directory
+2. Setup, using dynamic linking
+    - On Linux, `libshaderc_shared.so` will be searched in the `SHADERC_LIB_DIR`
+      environment variable or in the `/usr/lib/`
+    - On Linux, `/usr/lib/` will be automatically searched for `libshaderc_shared.so`
+      if `SHADERC_LIB_DIR` is not setted.
+    - On Windows (MSVC and gnu), If the `SHADERC_LIB_DIR` environment variable is set to
+      `/path/to/shaderc/bin/`, `shaderc_shared.dll` will be searched in the 
+      `/path/to/shaderc/bin/` directory
+3. Building from source
```
Comment (antiagainst) [inline README.md]: Can we delete this case? It's duplicating information with the following.

File: README.md (hunk 2 above)
Comment (antiagainst) [inline README.md]: I don't think we want ordered list: it's not indicating some sort of order; it's just different configurations.

File: README.md [hunk 8]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+
+shaderc library can be obtained from [here](https://github.com/google/shaderc#downloads)
+and linked, or can be [built from source](#building-from-source):
+
+1. Setup, using static linking
+    - On Linux, If the `SHADERC_LIB_DIR` environment variable is set to
+      `/path/to/shaderc/libs/`, it will take precedence and `libshaderc_combined.a`
+      (and the glsang and SPIRV libraries) will be searched in the
+      `/path/to/shaderc/libs/` directory. 
+    - On Linux, `/usr/lib/` will be automatically searched for `libshaderc_combined.a`
```
Comment (antiagainst) [inline README.md]: Otherwise, on Linux, ...

File: README.md [hunk 9]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+      if `SHADERC_LIB_DIR` is not setted.
```
Comment (antiagainst) [inline README.md]: s/setted/set/

File: README.md [hunk 10]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+      `/path/to/shaderc/libs/`, `shaderc_combined.lib` will be searched in the
+      `/path/to/shaderc/libs/` directory
+2. Setup, using dynamic linking
+    - On Linux, `libshaderc_shared.so` will be searched in the `SHADERC_LIB_DIR`
+      environment variable or in the `/usr/lib/`
+    - On Linux, `/usr/lib/` will be automatically searched for `libshaderc_shared.so`
+      if `SHADERC_LIB_DIR` is not setted.
+    - On Windows (MSVC and gnu), If the `SHADERC_LIB_DIR` environment variable is set to
+      `/path/to/shaderc/bin/`, `shaderc_shared.dll` will be searched in the 
+      `/path/to/shaderc/bin/` directory
```
Comment (antiagainst) [inline README.md]: Please end sentence with a period. :)

File: README.md [hunk 11]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+      `/path/to/shaderc/libs/` directory. 
+    - On Linux, `/usr/lib/` will be automatically searched for `libshaderc_combined.a`
+      if `SHADERC_LIB_DIR` is not setted.
+    - On Windows MSVC, If the `SHADERC_LIB_DIR` environment variable is set to
+      `/path/to/shaderc/libs/`, `shaderc_combined.lib` will be searched in the
+      `/path/to/shaderc/libs/` directory
+2. Setup, using dynamic linking
+    - On Linux, `libshaderc_shared.so` will be searched in the `SHADERC_LIB_DIR`
+      environment variable or in the `/usr/lib/`
+    - On Linux, `/usr/lib/` will be automatically searched for `libshaderc_shared.so`
```
Comment (antiagainst) [inline README.md]: Otherwise, on Linux, ...

File: README.md [hunk 12]
```diff
@@ -73,21 +73,46 @@ assert!(text_result.as_text().starts_with("; SPIR-V\n"));
...
+      if `SHADERC_LIB_DIR` is not setted.
```
Comment (antiagainst) [inline README.md]: s/setted/set/

**Review by antiagainst — COMMENTED** [review]
Thanks for the contribution and sorry about the delay!

**Comment by misha1sh** [comment]
Fixed. Thank you for code review.

**Review by antiagainst — APPROVED** [review]
Thanks!

**Pull request merged**